openssl/src/c_helpers.c

```diff
@@ -8,8 +8,12 @@ void rust_SSL_CTX_clone(SSL_CTX *ctx) {
     CRYPTO_add(&ctx->references,1,CRYPTO_LOCK_SSL_CTX);
 }
 
-void rust_EVP_PKEY_clone(EVP_PKEY *pkey) {
-    CRYPTO_add(&pkey->references,1,CRYPTO_LOCK_EVP_PKEY);
+EVP_PKEY* rust_EVP_PKEY_clone(EVP_PKEY *pkey) {
+    EVP_PKEY* new_pkey = EVP_PKEY_new();
+    if (EVP_PKEY_copy_parameters(new_pkey, pkey) != 0) {
```

Comment:
`EVP_PKEY_copy_parameters` looks to be a real function rather than a macro, so it can be defined in openssl-sys and then used directly from the rust Clone implementation. This C file only exists since `CRYPTO_add` is a macro.